### PR TITLE
MLIBZ-1604: Update a user causes 401 on subsequent API requests

### PR DESCRIPTION
**Pull request by thomasconner**
#### Description
This PR updates the active user data if a `PUT` request is made to update the user on the backend.

#### Changes
- Update active user data with data returned from `PUT` request made on the active user to the backend.
- Add unit tests for updating a user.

### CODE REVIEW

File: src/datastore/src/userstore.js [hunk 1]
```diff
@@ -51,7 +51,7 @@ class UserStore extends NetworkStore {
     }
 
     if (!data._id) {
-      return Promise.ject(new KinveyError('User must have an _id.'));
+      return Promise.reject(new KinveyError('User must have an _id.'));
```
Comment (tejasranade) [inline src/datastore/src/userstore.js]: A test to cover this error handling case would be nice.

File: test/unit/entity/user.test.js [hunk 2]
```diff
@@ -300,4 +301,62 @@ describe('User', function() {
...
+          const responseData = assign(requestData, { _kmd: { authtoken: randomString() } });
+
+          // Kinvey API response
+          nock(this.client.apiHostname, { encodedQueryParams: true })
+            .put(`${user.pathname}/${user._id}`, requestData)
+            .reply(200, responseData);
+
+          return user.update({ email: email })
+            .then(() => {
+              expect(user.data).toEqual(responseData);
```
Comment (tejasranade) [inline test/unit/entity/user.test.js]: Does the updated user object, in this case, contain the auth token? I'd assume it doesn't. In either case, can we add an assert for the correct behavior?

File: src/datastore/src/userstore.js (hunk 1 above)
Comment (thomasconner) [inline src/datastore/src/userstore.js]: A [unit test](https://github.com/Kinvey/js-sdk/pull/101/files#diff-ea2fab4905bfc00b94d59bb6949998e2R306) has already been written to cover this case.

Comment (tejasranade) [inline src/datastore/src/userstore.js]: Oops 😃

File: test/unit/entity/user.test.js (hunk 2 above)
Comment (thomasconner) [inline test/unit/entity/user.test.js]: This [line](https://github.com/Kinvey/js-sdk/pull/101/files#diff-ea2fab4905bfc00b94d59bb6949998e2R323) is where the `authtoken` is recreated. This [line](https://github.com/Kinvey/js-sdk/pull/101/files#diff-ea2fab4905bfc00b94d59bb6949998e2R333) verifies the active user contains the new data including the new `authtoken`.

File: src/entity/src/user.js [hunk 3]
```diff
@@ -691,18 +691,17 @@ export default class User {
...
     return store.update(data, options)
-      .then(() => {
-        this.data = data;
-        return this.isActive();
-      })
-      .then((isActive) => {
-        if (isActive) {
-          return CacheRequest.setActiveUser(this.client, this.data);
+      .then((data) => {
+        if (this.isActive() === true) {
```
Comment (thomasconner) [inline src/entity/src/user.js]: Yes. I will remove.

**Pull request merged**